lib/cadet/assessments/question.ex

```diff
@@ -1,34 +1,37 @@
 defmodule Cadet.Assessments.Question do
   @moduledoc """
-  Questions model contains domain logic for questions management 
+  Questions model contains domain logic for questions management
   including programming and multiple choice questions.
   """
   use Cadet, :model
 
   alias Cadet.Assessments.Assessment
-  alias Cadet.Assessments.ProblemType
+  alias Cadet.Assessments.QuestionType
   alias Cadet.Assessments.QuestionTypes.ProgrammingQuestion
   alias Cadet.Assessments.QuestionTypes.MCQQuestion
+  alias Cadet.Assessments.QuestionTypes.Library
 
   schema "questions" do
     field(:title, :string)
     field(:display_order, :integer)
-    field(:weight, :integer)
     field(:question, :map)
-    field(:type, ProblemType)
+    field(:type, QuestionType)
     field(:raw_question, :string, virtual: true)
+    field(:max_xp, :integer)
+    embeds_one(:library, Library)
     belongs_to(:assessment, Assessment)
     timestamps()
   end
 
-  @required_fields ~w(title weight question type)a
-  @optional_fields ~w(display_order raw_question)a
+  @required_fields ~w(title question type assessment_id)a
+  @optional_fields ~w(display_order raw_question max_xp)a
```

Comment:
`library` and `:raw_library`

Reply:
these shouldn't be here. `cast/4` doesn't work with embeds.

Reply:
I thought you need to cast `:library` first before you can `cast_embed/3`?

Reply:
Cos `cast_embed/3` is getting its data from `%Changeset{data: data}`

Reply:
You can try it out in the REPL `iex -S mix`

Reply:
Never mind, I stand corrected

Reply:
Also no need to add it to the fields. can check with this:

```
params = %{"title": "hello", "assessment_id": 4, "type": :multiple_choice, "question": %{"content": "hello", choices: [%{"content": "world", "choice_id": 0, "is_correct": true}]}, "library": %{"version": 0, "globals": ["hello", "world"]}}

Cadet.Assessments.Question.changeset(%Cadet.Assessments.Question{}, params)
```

Reply:
Yup, I derped. Sorry :/